lib/lint/collectUsed/fileCollector.js

```diff
@@ -0,0 +1,21 @@
+module.exports = function collectFiles(flow, callback){
+  var stack = [flow.files.queue[0]];
+  var collectedFiles = {};
+  var handled = {};
+  var cursor;
+
+  while (cursor = stack.pop())
+  {
+    // mark file as handled
+    handled[cursor.uniqueId] = true;
```

Comment:
I thought about Map, but [Map is too slow](https://jsperf.com/object-vs-array-vs-set-vs-map)